.github/workflows/ci.yml

```diff
@@ -12,120 +12,82 @@ on:
     # * is a special character in YAML so you have to quote this string
     - cron: '0 9 * * 1'
 
-env:
-  CRDS_PATH: $HOME/crds_cache
-  CRDS_CLIENT_RETRY_COUNT: 3
-  CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
-
 jobs:
-  style:
-    name: Code style checks
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
-      - uses: actions/cache@v3
-        with:
-          path: ${{ env.pythonLocation }}
-          key: style-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
-      - run: pip install ruff
-      - run: ruff src
-  audit:
-    name: Bandit security audit
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
-      - uses: actions/cache@v3
-        with:
-          path: ${{ env.pythonLocation }}
-          key: audit-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
-      - run: pip install bandit
-      - run: bandit -r -ll src
   test:
-    name: test
-    needs: [ style, audit ]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - linux: check-style
+        - linux: check-security
+        - linux: check-build
+        - linux: test-xdist
+          python-version: '3.8'
+        - macos: test-xdist
+          python-version: '3.8'
+        - linux: test-xdist
+          python-version: '3.9'
+        - macos: test-xdist
+          python-version: '3.9'
+        - linux: test-xdist
+          python-version: '3.10'
+        - macos: test-xdist
+          python-version: '3.10'
+        - linux: test-xdist
+          python-version: '3.11'
+        - macos: test-xdist
+          python-version: '3.11'
+        - linux: test-numpy122
+          python-version: '3.10'
+        - linux: test-numpy121
+          python-version: '3.10'
+        - linux: test-numpy122
+          python-version: '3.9'
+        - linux: test-numpy121
+          python-version: '3.9'
+        - linux: test-numpy120
+          python-version: '3.9'
+        - linux: test-numpy122
+          python-version: '3.8'
+        - linux: test-numpy121
+          python-version: '3.8'
+        - linux: test-numpy120
+          python-version: '3.8'
+        - linux: test-cov
+          coverage: 'codecov'
```

Comment:
I suggest breaking these into smaller sections. Like a `numpy` section or a `check` section. This will make browsing the CI easier to do from the CI overview.

Reply:
Something like the `asdf-astropy` CI: https://github.com/astropy/asdf-astropy/blob/main/.github/workflows/ci.yml